x-pack/plugins/infra/server/utils/route_validation.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type {
```

Comment:
a lot of this was copied from a couple other plugins which seem to have copied from each other.  i had to change it to handle the ExactType.  I may open an issue to see if we can get this into core.